src/test/java/picard/sam/MergeBamAlignmentTest.java

```diff
@@ -1930,5 +1931,73 @@ public void testHeaderFromMappedBreaks(final String filename) throws IOException
+        final SAMRecordSetBuilder samRecordSetBuilderUnmappedBam = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.queryname);
+        samRecordSetBuilderUnmappedBam.setRandomSeed(12345);
+        final SAMFileHeader header = samRecordSetBuilderUnmappedBam.getHeader();
+        header.setSequenceDictionary(SAMSequenceDictionaryExtractor.extractDictionary(fasta.toPath()));
+
+        final SAMRecordSetBuilder samRecordSetBuilderAlignedBam = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.queryname);
+        samRecordSetBuilderUnmappedBam.setRandomSeed(12345);
+        final SAMFileHeader headerAligned = samRecordSetBuilderUnmappedBam.getHeader();
+        headerAligned.setSequenceDictionary(SAMSequenceDictionaryExtractor.extractDictionary(fasta.toPath()));
+
+        if (!paired) {
+            samRecordSetBuilderUnmappedBam.addFrag("theRead", 1, 1, false, false, null, null, -1);
+            samRecordSetBuilderAlignedBam.addFrag("theRead", 1, 1, false, false, null, null, -1);
+        } else {
+            samRecordSetBuilderUnmappedBam.addPair("theRead", 1, 1, 65, firstUnMapped, secondUnMapped, null, null, false, true, -1);
+            samRecordSetBuilderAlignedBam.addPair("theRead", 1, 1, 65, false, false, null, null, false, true, -1);
+        }
+
+        //add to both unmappedSam and alignedSam
+        final File unmappedSam = File.createTempFile("unmapped.", ".sam");
+        unmappedSam.deleteOnExit();
+        final File alignedSam = File.createTempFile("aligned.", ".sam");
+        alignedSam.deleteOnExit();
+
+        final SAMFileWriterFactory factory = new SAMFileWriterFactory();
+
+        final SAMFileWriter unmappedWriter = factory.makeSAMWriter(header, false, unmappedSam);
+        final SAMFileWriter alignedWriter = factory.makeSAMWriter(header, false, alignedSam);
+        for (final SAMRecord rec : samRecordSetBuilderUnmappedBam.getRecords()) {
+            unmappedWriter.addAlignment(rec);
+        }
+
+        for (final SAMRecord rec : samRecordSetBuilderAlignedBam.getRecords()) {
+            alignedWriter.addAlignment(rec);
+        }
+        unmappedWriter.close();
+        alignedWriter.close();
+
+        //run merge
+        final File output = File.createTempFile("output", ".sam");
+        output.deleteOnExit();
+        final List<String> args = new ArrayList<>(Arrays.asList(
+                "UNMAPPED_BAM=" + unmappedSam.getAbsolutePath(),
+                "ALIGNED_BAM=" + alignedSam.getAbsolutePath(),
+                "OUTPUT=" + output.getAbsolutePath(),
+                "REFERENCE_SEQUENCE=" + fasta.getAbsolutePath())
+        );
+
+        //should throw PicardException when run because mapped read found in UNMAPPED_BAM input
+        if (shouldThrowException) {
+            Assert.assertThrows(PicardException.class, () -> runPicardCommandLine(args));
```

Comment:
this didn't help...

refactor the insides of the test into a private static method, then have two tests call it, one which expects an exception and one that doesn't.

Reply:
I may be confused about your comment, but I think the additional test case, where all reads are unmapped and `shouldThrowException` is set to false, does distinguish between properly failing and always failing.  The test no longer expects an exception through the annotation, so it is testing that an exception is thrown when it should be, and not when it's shouldn't.  
But I will refactor as requested, seems like a more elegant solution anyway.

Reply:
upon second read, I agree that your solution works... :-) sorry for the confusion.